helper/terraformtype/helper/schema/attributes.go

```diff
@@ -59,3 +60,42 @@ func ParseAttributeReference(reference string) ([]string, error) {
 
 	return attributeReferenceParts, nil
 }
+
+// ValidateAttributeReference validates schema attribute reference.
+// Attribute references are used in Schema fields such as AtLeastOneOf, ConflictsWith, and ExactlyOneOf.
+func ValidateAttributeReference(tfresource *tfschema.Resource, reference string) ([]string, error) {
+	var curSchemaOrResource interface{} = tfresource
+	attributeReferenceParts := strings.Split(reference, ".")
+	for idx, attributeReferencePart := range attributeReferenceParts {
+		attributeReference := strings.Join(attributeReferenceParts[:idx+1], ".")
+		if math.Mod(float64(idx), 2) == 1 {
+			// For odd part, ensure it is a 0 and the containing schema has `MaxItems` set to `1`. This is because
+			// reference among multiple instances of the same nested block is not supported in current plugin SDK.
+			attributeReferencePartInt, err := strconv.Atoi(attributeReferencePart)
+
+			configurationBlockReferenceErr := fmt.Errorf("%q configuration block attribute references must be separated by .0", attributeReference)
+			if err != nil {
+				return nil, configurationBlockReferenceErr
+			}
+			if attributeReferencePartInt != 0 {
+				return nil, configurationBlockReferenceErr
+			}
+
+			curSchema := curSchemaOrResource.(*tfschema.Schema)
+			if curSchema.MaxItems != 1 || curSchema.Type != tfschema.TypeList {
+				return nil, fmt.Errorf("%q configuration block attribute references are only valid for TypeList and MaxItems: 1 attributes", attributeReference)
+			}
+			curSchemaOrResource = curSchema.Elem
```

Comment:
Please introduce type assertion safety here with a `switch` statement utilizing `.(type)`:

```suggestion
			switch current := curSchemaOrResource.(type) {
			case *tfschema.Schema:
				if current.MaxItems != 1 || current.Type != tfschema.TypeList {
					return nil, fmt.Errorf("%q configuration block attribute references are only valid for TypeList and MaxItems: 1 attributes", attributeReference)
				}
				curSchemaOrResource = current.Elem
			}
```

Reply:
Sure. Also for the default case (which means we are not capable to deal with for now), I think we shall just return `nil` to avoid false positive.